.github/workflows/test.yml

```
@@ -87,16 +247,48 @@ jobs:
   integration-test-postgres:
     name: Run Integration Tests (on Postgres)
     runs-on: ubuntu-latest
+    needs: container-test-builds
     steps:
       - name: Checkout
         uses: actions/checkout@v4
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 20
+          check-latest: true
+          cache: "npm"
+          cache-dependency-path: "**/package-lock.json"
+
+      - name: Setup NuGet
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: 8.x
+
       - name: Install script dependencies
         run: npm install --prefix ./.ci
+
+      - name: Download Docker images
```

```suggestion
      - name: Download cached Docker images
```

Same for the other occurrences